### PR TITLE
Modify Deployment to Allow High Availability

### LINKED ISSUE #251: HA deployments with pod disruption budget
### Description
The Karydia Helm chart hard-codes `1` replica. As it registers webhooks that trigger for `pods` on `CREATE`/`UPDATE` operations, once the single Karydia pod is deleted, the system can never recover itself automatically.
Webhooks on CREATE/UPDATE for pods is dangerous in general if the server runs inside the cluster itself. So, what you should at least do is deploy the webhook server in high-availability mode (>= 3 replicas with pod anti-affinity) and with pod disruption budgets to ensure that at least one server is always reachable (to allow recreating the others).

### PR DESCRIPTION
**Pull request by Neumann-Nils**
### Description
This PR introduces three changes to ensure an high available deployment for Karydia:
1. Increase the replica count to `3`
2. Define pod anti-affinity
3. Define pod disruption budget

[Fixes #251]

### Checklist
Before submitting this PR, please make sure:
- [x] your code builds clean with `make`
- [x] your code lets succeed unit tests with `make test`
- [x] your code lets succeed integration tests>

### CODE REVIEW

File: install/charts/templates/deployment.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ metadata:
     app: {{ .Values.metadata.labelApp }}
   namespace: {{ .Release.Namespace }}
 spec:
-  replicas: 1
+  replicas: 3
```
Comment (rfranzke) [inline install/charts/templates/deployment.yaml]: I would make this configurable in the Helm chart.

File: install/charts/templates/deployment.yaml [hunk 2]
```diff
@@ -32,6 +32,18 @@ spec:
...
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 100
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                - key: app
+                  operator: In
+                  values:
+                  - {{ .Values.metadata.labelApp }}
+              topologyKey: "node"
```
Comment (rfranzke) [inline install/charts/templates/deployment.yaml]: What happens if there is only one node? Will only one replica start and the others remain pending? I think we should achieve the following: If there is a node for each replica then distribute them. If there are less nodes then still start all configured replicas (even if this means that on some nodes there is > 1 replica).

File: install/charts/templates/deployment.yaml (hunk 1 above)
Comment (Neumann-Nils) [reply install/charts/templates/deployment.yaml]: Done.

File: install/charts/templates/deployment.yaml (hunk 2 above)
Comment (Neumann-Nils) [reply install/charts/templates/deployment.yaml]: In my understanding, you are describing what the `preferredDuringSchedulingIgnoredDuringExecution` should do.

As written in [docs](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity): "`preferredDuringSchedulingIgnoredDuringExecution` would be try to run this set of pods in failure zone XYZ, but if it’s not possible, then allow some to run elsewhere."

As far as I tested it, the Karydia pods will be spread over multiple nodes when possible. But if i.e. only one node is available, all three replicas will be deployed on the same node.

Comment (rfranzke) [reply install/charts/templates/deployment.yaml]: Okay, thanks, great!

Comment (rfranzke) [reply install/charts/templates/deployment.yaml]: Okay, thanks, great!

File: Makefile [hunk 3]
```diff
@@ -49,7 +49,9 @@ container-dev:
 
 .PHONY: deploy-dev
 deploy-dev:
-	kubectl cp bin/karydia -n karydia $(shell kubectl get pods -n=karydia --selector=app=karydia --output=jsonpath='{.items[0].metadata.name}'):/usr/local/bin/karydia$(DEV_POSTFIX)
+	for i in $(shell kubectl get pods -n=karydia --selector=app=karydia --output=jsonpath='{.items[*].metadata.name}'); do \
+		kubectl cp bin/karydia -n karydia $$i:/usr/local/bin/karydia-dev; \
```
Comment (dacappo) [inline Makefile]: `-dev` hardcoded vs `$(DEV_POSTFIX)`

Comment (Neumann-Nils) [reply Makefile]: Thanks, looks like I forgot that.

**Pull request merged**